api/v1alpha1/clusterissuer_types.go

```diff
@@ -17,22 +17,40 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/cert-manager/issuer-lib/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:resource:scope=Cluster
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].status"
+// +kubebuilder:printcolumn:name="Reason",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].reason"
+// +kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].message"
+// +kubebuilder:printcolumn:name="LastTransition",type="string",type="date",JSONPath=".status.conditions[?(@.type==\"Ready\")].lastTransitionTime"
+// +kubebuilder:printcolumn:name="ObservedGeneration",type="integer",JSONPath=".status.conditions[?(@.type==\"Ready\")].observedGeneration"
+// +kubebuilder:printcolumn:name="Generation",type="integer",JSONPath=".metadata.generation"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
 
 // ClusterIssuer is the Schema for the clusterissuers API
 type ClusterIssuer struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   IssuerSpec   `json:"spec,omitempty"`
-	Status IssuerStatus `json:"status,omitempty"`
+	Spec   IssuerSpec            `json:"spec,omitempty"`
+	Status v1alpha1.IssuerStatus `json:"status,omitempty"`
+}
+
+func (vi *ClusterIssuer) GetStatus() *v1alpha1.IssuerStatus {
+	return &vi.Status
 }
 
+func (vi *ClusterIssuer) GetIssuerTypeIdentifier() string {
+	return "clusterissuers.sample-issuer.example.com"
+}
+
+var _ v1alpha1.Issuer = &ClusterIssuer{}
```

Comment:
suggestion: I think a lot of people using sample-external-issuer might be pretty new to Go based on what we've seen in the past. I like that we have an assertion here that ClusterIssuer satisfies the interface, but this Go idiom is pretty difficult to work out if you're new (and it's hard to Google it too).

Could we add a comment to make it clear why this is here? We don't need to go full handholding with this repo but I think this might be worth it!